### PR TITLE
build process updates, deprecate docker-launcher in favour of docker-compose

### PR DESCRIPTION
**Pull request by tkfu**

### CODE REVIEW

**Comment by tkfu** [comment]
Note: the docker-compose file that includes rvi_sota_client is using the image in our repo, which is 7 months old. I wasn't able to get the client to build in my VM, so for now the YAML refers to that out of date version.

File: deploy/docker-compose/server-rvi-client.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  resolver:
+    image: advancedtelematic/sota-resolver
+    ports:
+      - 8081:8081
+    expose:
+      - 8081
+    depends_on:
+      - mysql
+      - rvi_server
```
Comment (jerrytrieu) [inline deploy/docker-compose/server-rvi-client.yml]: Don't need rvi in resolver.

File: deploy/docker-compose/server-rvi-client.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  core:
+    image: advancedtelematic/sota-core
+    ports:
+      - 8080:8080
+    expose:
+      - 8080
+    depends_on:
+      - mysql
+      - rvi_server
```
Comment (jerrytrieu) [inline deploy/docker-compose/server-rvi-client.yml]: Need resolver here.

File: deploy/docker-compose/server-rvi-client.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    environment:
+      HOST: 0.0.0.0
+      CORE_DB_URL: "jdbc:mariadb://mysql:3306/sota_core"
+      CORE_DB_MIGRATE: 'true'
+      RESOLVER_API_URI: "http://resolver:8081"
+      RVI_URI: "http://rvi_server:8801"
```
Comment (jerrytrieu) [inline deploy/docker-compose/server-rvi-client.yml]: Need SOTA_SERVICES_URI env, which needs to be ip accessible from rvi-server container, which is a pain.

File: deploy/docker-compose/server-rvi-client.yml (hunk 1 above)
Comment (tkfu) [inline deploy/docker-compose/server-rvi-client.yml]: This is just a hack. The whole thing fails if core or resolver start running migrations before MariaDB is fully up and running. `depends_on` just means "don't start this container until all of its dependencies have started"; it's not synonymous with "links" from docker-compose v.1.

File: deploy/docker-compose/server-rvi-client.yml (hunk 2 above)
Comment (tkfu) [inline deploy/docker-compose/server-rvi-client.yml]: That would create a circular dependency. In docker-compose v.2, by default all the containers are on a network and can address each other by name. (you can explicitly define other topologies, but that's the default)

**Pull request merged**